x-pack/plugins/security_solution/public/detection_engine/rule_management_ui/components/rules_table/add_prebuilt_rules_table/add_prebuilt_rules_table_context.tsx

```diff
@@ -44,9 +45,14 @@ export interface AddPrebuiltRulesTableState {
    */
   isFetched: boolean;
   /**
-   * Is true whenever a background refetch is in-flight, which does not include initial loading
+   * Is true when doing an initial fetch of rules to install
    */
-  isRefetching: boolean;
+  shouldShowLoadingOverlay: boolean;
+  /**
+   * Is true when installing security_detection_rules package in background
+   * or refetching rules to install rules in the background
+   */
+  shouldShowLinearProgress: boolean;
```

Comment:
@xcrzx  Should these be renamed to something more generic, unrelated to the UI?

Maybe `isFetchingRules` and `isRefetchingRulesOrInstallingPackage`?

Reply:
++ Yes, the context should not be aware of linear progress or loading overlay. Something like `isFetchingRules` and `isFetchingRulesPackage` sounds generic enough.